src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/GenAPIConfiguration.cs

```diff
@@ -0,0 +1,317 @@
+        {
+            if (assemblyReferencesPaths == null)
+            {
+                return this;
+            }
+            foreach (string path in assemblyReferencesPaths)
+            {
+                ThrowIfFileSystemEntryNotFound(nameof(assemblyReferencesPaths), path);
+            }
+            _assemblyReferencesPaths = assemblyReferencesPaths;
+            return this;
+        }
+
+        public Builder WithAllowUnsafe(bool allowUnsafe)
+        {
+            _allowUnsafe = allowUnsafe;
+            return this;
+        }
+
+        public Builder WithRespectInternals(bool respectInternals)
+        {
+            _respectInternals = respectInternals;
+            return this;
+        }
+
+        public GenAPIConfiguration Build()
+        {
+            AssemblySymbolLoader loader;
+            Dictionary<string, IAssemblySymbol> assemblySymbols;
+
+            if (_assembliesPaths?.Length > 0)
+            {
+                bool resolveAssemblyReferences = _assemblyReferencesPaths?.Count() > 0;
+                loader = new AssemblySymbolLoader(resolveAssemblyReferences, _respectInternals);
+                if (_assemblyReferencesPaths?.Length > 0)
+                {
+                    loader.AddReferenceSearchPaths(_assemblyReferencesPaths);
+                }
+                assemblySymbols = new Dictionary<string, IAssemblySymbol>(loader.LoadAssembliesAsDictionary(_assembliesPaths));
+            }
+            else if (_assemblyTexts?.Count() > 0)
+            {
+                loader = new AssemblySymbolLoader(resolveAssemblyReferences: true, includeInternalSymbols: _respectInternals);
+                loader.AddReferenceSearchPaths(typeof(object).Assembly!.Location!);
+                loader.AddReferenceSearchPaths(typeof(DynamicAttribute).Assembly!.Location!);
+
+                assemblySymbols = new Dictionary<string, IAssemblySymbol>();
+                foreach ((string assemblyName, string assemblyText) in _assemblyTexts)
+                {
+                    using Stream assemblyStream = EmitAssemblyStreamFromSyntax(assemblyText, enableNullable: true, allowUnsafe: _allowUnsafe, assemblyName: assemblyName);
+                    if (loader.LoadAssembly(assemblyName, assemblyStream) is IAssemblySymbol assemblySymbol)
+                    {
+                        assemblySymbols.Add(assemblyName, assemblySymbol);
+                    }
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException("No assemblies were specified, either from files or from streams.");
+            }
+
+            return new GenAPIConfiguration()
+            {
+                Loader = loader,
+                AssemblySymbols = assemblySymbols
+            };
+        }
+
+
+        private static IEnumerable<KeyValuePair<string, ReportDiagnostic>> DiagnosticOptions { get; } = new[]
```

Comment:
All the code under this line is duplicate from a test extension class. I need to consolidate it.